src/mame/rm/rmnimbus_m.cpp

```diff
@@ -1075,6 +1072,28 @@ void rmnimbus_state::mouse_js_reset()
 	m_nimbus_mouse.m_mouse_timer->adjust(attotime::zero, 0, attotime::from_hz(MOUSE_POLL_FREQUENCY));
 }
 
+uint8_t rmnimbus_state::nimbus_joystick_r()
+{
+	/* Only the joystick drection data is read from this port
+	   (which corresponds to the the low nibble of m_io_joystick0).
+	   The joystick buttons are read from the mouse data port instead. */
+	uint8_t result = m_io_joystick0->read() & 0x0f;
+
+	if (result & CONTROLLER_RIGHT)
+	{
+		// when the stick is pushed right the left bit must also be set!
+		result |= CONTROLLER_LEFT;
+	}
+
+	if (result & CONTROLLER_UP)
+	{
+		// when the stick is pushed up the down bit must also be set!
+		result |= CONTROLLER_DOWN;
+	}
```

Comment:
This seems like a very odd design decision, and at first glance it would seem to require more logic than just mapping one direction to each bit.  Do you know why it has this quirk of always setting the left bit for horizontal directions and always setting the down bit for vertical directions?

Reply:
I'm guessing it's related to the physical hardware design, but it's actually stated in the service manual. They have a table saying something like "right = left + right" :-) Maybe it's really a bit to say the X axis is effected and then a bit for positive/negative.

It threw me at first because without setting the extra bits only left and down worked. Then I noticed the cryptic table in the service manual.

All 8 directions work with my RM basic test app so fairly confident it's correct now!